x/yarpctest/api/testing.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package api
+
+import (
+	"sync"
+	"testing"
+)
+
+// TestingT is an interface wrapper around *testing.T and *testing.B
+type TestingT interface {
```

Comment:
Is there a strong reason to use this instead of `testing.TB` directly? In the future, we won't be able to modify this interface anyways since that would end up a breaking change.

Reply:
I was under the impression that it was better practice to define your own.  That might have been more for internal/non-exported test packages though, since it gives more flexibility when you might need to make a breaking change.  Totally flexible on this though.

Reply:
I'd say since `testing.TB` is a common stdlib type, just use that, but I'm also fine with either.

Reply:
You’re both right. I think testing.TB is solid as ice at this point though. Your call.